client.go

```diff
@@ -523,15 +523,17 @@ func (c *Client) Join(channels ...string) {
 
 	// If we have an active connection, explicitly join
 	// before we add the joined channels to our map
+	connActive := c.connActive.get()
```

Comment:
not an expert, but doesn't this kinda defeat the purpose of the atomic bool?

Reply:
Should we be checking if the connection is active for each join message and each channel in the map then?

Reply:
hmm no nevermind. This might be fine here

Reply:
You are right @gempir, it does defeat the purpose of the atomic bool. There is a chance that inbetween the connActive check, the JOIN succeeds and initialJoins has been called.